src/Diagnostics.RuntimeHost/Controllers/DiagnosticControllerBase.cs

```diff
@@ -220,25 +220,27 @@ private DiagnosticApiResponse CreateQueryExceptionResponse(Exception ex, Definit
             return DiagnosticApiResponse.FromCsxResponse(response, dataProvidersMetadata);
         }
 
-        protected async Task<IActionResult> GetInsights(TResource resource, string supportTopicId, string minimumSeverity, string startTime, string endTime, string timeGrain)
+        protected async Task<IActionResult> GetInsights(TResource resource, string pesId, string supportTopicId, string startTime, string endTime, string timeGrain)
         {
             if (!DateTimeHelper.PrepareStartEndTimeWithTimeGrain(startTime, endTime, timeGrain, out DateTime startTimeUtc, out DateTime endTimeUtc, out TimeSpan timeGrainTimeSpan, out string errorMessage))
             {
                 return BadRequest(errorMessage);
             }
 
-            OperationContext<TResource> cxt = PrepareContext(resource, startTimeUtc, endTimeUtc, forceInternal: true);
+            var supportTopic = new SupportTopic() { Id = supportTopicId, PesId = pesId };
+            OperationContext<TResource> cxt = PrepareContext(resource, startTimeUtc, endTimeUtc, true, supportTopic);
 
             List<AzureSupportCenterInsight> insights = null;
             string error = null;
             List<Definition> detectorsRun = new List<Definition>();
+            IEnumerable<Definition> allDetectors = null;
             try
             {
                 supportTopicId = ParseCorrectSupportTopicId(supportTopicId);
-                var allDetectors = (await ListDetectorsInternal(cxt)).Select(detectorResponse => detectorResponse.Metadata);
+                allDetectors = (await ListDetectorsInternal(cxt)).Select(detectorResponse => detectorResponse.Metadata);
 
                 var applicableDetectors = allDetectors
-                    .Where(detector => string.IsNullOrWhiteSpace(supportTopicId) || detector.SupportTopicList.FirstOrDefault(supportTopic => supportTopic.Id == supportTopicId) != null);
+                    .Where(detector => string.IsNullOrWhiteSpace(supportTopicId) || detector.SupportTopicList.FirstOrDefault(st => st.Id == supportTopicId) != null);
```

Comment:
you probably need to match PesId too also now for applicableDetectors

Reply:
Our filtering was taking care of this right? Right now I don't have pesId piped up completely. Nothing should have changed in this part of the logic.

Reply:
Filtering will take care it in general. 
Rare scenario : if you have a detector marked for windows and linux , but support topic is only for windows, you dont want to trigger this detector in ASC for linux support topic (with same id)